rdr_service/dao/questionnaire_response_dao.py

```diff
@@ -349,6 +368,18 @@ def _validate_link_ids_from_resource_json_group(self, resource, link_ids):
                     if "answer" in section:
                         logging.error(f'Questionnaire response contains invalid link ID "{link_id}"')
 
+    @staticmethod
+    def _get_module_from_questionnaire_history(questionnaire_history):
+        """ Extract the module code value from a questionnaire_history result JSON data (resource field) """
+
+        # Unittest/lower environments may not have expected questionnaire_history content, so allow for missing data
+        result = None
+        history_data = json.loads(questionnaire_history.resource) if questionnaire_history.resource else {}
+        if 'group' in history_data.keys() and 'concept' in history_data['group'].keys():
+            concepts = history_data['group']['concept']
+            result = concepts[0]['code'] if len(concepts) else None
+        return result
```

Comment:
It was breaking in local database /  unittest database setups which did not have conformant `questionnaire_history` data and I was trying to get it working as fast as possible because of the time pressure on getting this into production.

Reply:
@kenny-skaggs I am working on a refactor.   Can we move forward with this PR as is if there are no other issues, and I will make sure there's a backlog ticket to adapt the way the code value is retrieved?  I think I will also look at optimizing the way the profile update code list is generated via the code cache.